src/librbd/cache/rwl/LogOperation.h

```diff
@@ -0,0 +1,183 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_LIBRBD_CACHE_RWL_LOG_OPERATION_H
+#define CEPH_LIBRBD_CACHE_RWL_LOG_OPERATION_H
+
+#include "include/utime.h"
+#include "librbd/cache/rwl/LogEntry.h"
+#include "librbd/cache/rwl/SyncPoint.h"
+
+namespace librbd {
+namespace cache {
+namespace rwl {
+struct WriteBufferAllocation;
+
+class WriteLogOperationSet;
+
+class WriteLogOperation;
+
+class GenericWriteLogOperation;
+
+class SyncPointLogOperation;
+
+class GenericLogOperation;
+
+using GenericLogOperationSharedPtr = std::shared_ptr<GenericLogOperation>;
+
+using GenericLogOperationsVector = std::vector<GenericLogOperationSharedPtr>;
+
+class GenericLogOperation {
+protected:
+  PerfCounters *m_perfcounter = nullptr;
+public:
+  utime_t dispatch_time;         // When op created
+  utime_t buf_persist_time;      // When buffer persist begins
+  utime_t buf_persist_comp_time; // When buffer persist completes
+  utime_t log_append_time;       // When log append begins
+  utime_t log_append_comp_time;  // When log append completes
+  GenericLogOperation(const utime_t dispatch_time, PerfCounters *perfcounter);
+  virtual ~GenericLogOperation() { };
+  GenericLogOperation(const GenericLogOperation&) = delete;
+  GenericLogOperation &operator=(const GenericLogOperation&) = delete;
+  virtual std::ostream &format(std::ostream &os) const;
+  friend std::ostream &operator<<(std::ostream &os,
+                                  const GenericLogOperation &op);
+  virtual const std::shared_ptr<GenericLogEntry> get_log_entry() = 0;
+  virtual const std::shared_ptr<WriteLogEntry> get_write_log_entry() {
+    return nullptr;
+  }
+  virtual void appending() = 0;
+  virtual void complete(int r) = 0;
+  virtual GenericWriteLogOperation *get_gen_write_op() {
```

Comment:
Nit: drop this method and replace w/ a generic method (to increase the metrics in this case)

Reply:
Done